public/app/core/services/backend_srv.ts

```diff
@@ -84,100 +46,33 @@ export class BackendSrv implements BackendService {
     }
   }
 
-  async get<T = any>(url: string, params?: any, requestId?: string): Promise<T> {
-    return await this.request({ method: 'GET', url, params, requestId });
-  }
-
-  async delete(url: string) {
-    return await this.request({ method: 'DELETE', url });
-  }
-
-  async post(url: string, data?: any) {
-    return await this.request({ method: 'POST', url, data });
-  }
-
-  async patch(url: string, data: any) {
-    return await this.request({ method: 'PATCH', url, data });
-  }
-
-  async put(url: string, data: any) {
-    return await this.request({ method: 'PUT', url, data });
-  }
-
-  withNoBackendCache(callback: any) {
-    this.noBackendCache = true;
-    return callback().finally(() => {
-      this.noBackendCache = false;
-    });
+  async request<T = any>(options: BackendSrvRequest): Promise<T> {
+    return this.fetch<T>(options)
+      .pipe(map((response: FetchResponse<T>) => response.data))
+      .toPromise();
   }
 
-  requestErrorHandler = (err: ErrorResponse) => {
-    if (err.isHandled) {
-      return;
-    }
-
-    let data = err.data ?? { message: 'Unexpected error' };
-    if (typeof data === 'string') {
-      data = { message: data };
-    }
-
-    if (err.status === 422) {
-      this.dependencies.appEvents.emit(AppEvents.alertWarning, ['Validation failed', data.message]);
-      throw data;
-    }
-
-    if (data.message) {
-      let description = '';
-      let message = data.message;
-      if (message.length > 80) {
-        description = message;
-        message = 'Error';
-      }
-
-      this.dependencies.appEvents.emit(err.status < 500 ? AppEvents.alertWarning : AppEvents.alertError, [
-        message,
-        description,
-      ]);
-    }
-
-    throw data;
-  };
-
-  async request(options: BackendSrvRequest): Promise<any> {
-    // A requestId is a unique identifier for a particular query.
-    // Every observable below has a takeUntil that subscribes to this.inFlightRequests and
-    // will cancel/unsubscribe that observable when a new datasourceRequest with the same requestId is made
+  fetch<T>(options: BackendSrvRequest): Observable<FetchResponse<T>> {
```

Comment:
I've considered this a matter of taste for a long time: Since `T` only occurs in the return, it's no better than asserting.

But it just struck me that a return type of `unknown` instead of a "dumb generic" (i.e. it's just passed through) is much more suitable because usually, generics indicate some sort of internal logic. But when used like this, there obviously is none.
`any` could of course be used as well instead of `unknown`, it would just depend on whether we want to force the user to assert the return value in order to access props on it.

As an aside - might it be worthwhile to call it `observableFetch` or `fetchObservable`, to highlight the difference to whatwg fetch?

Reply:
> Since T only occurs in the return, it's no better than asserting.

You mean typecasting the result? Think this is cleaner. 

> As an aside - might it be worthwhile to call it observableFetch or fetchObservable, to highlight the difference to whatwg fetch?

yea, not sure, we have request already which is exactly the as fetch except returns Promise

Reply:
> You mean typecasting the result?

It's not called a typecast in TS because it doesn't actually cast the type. (i.e. it only changes the type at compile time but doesn't affect run time)

> Think this is cleaner.

How come?

> we have request already which is exactly the as fetch except returns Promise

`observableRequest`? ;)